backends/bmv2/jsonconverter.cpp

```diff
@@ -2214,13 +2214,12 @@ void JsonConverter::addMetaInformation() {
   toplevel.emplace("__meta__", meta);
 }
 
-void JsonConverter::convert(P4::ReferenceMap* refMap, P4::TypeMap* typeMap,
-                            const IR::ToplevelBlock* toplevelBlock,
-                            P4::ConvertEnums::EnumMapping* enumMap) {
+void JsonConverter::convert(BMV2::MidEnd& midend,
```

Comment:
Frankly this looks weird; why should the JsonConverter depend on the structure of the BMV2::MidEnd?

Also, it is not clear that we always want to pass the MidEnd reference map; it is up to the caller to decide that, not the callee.